tasks/InsightAppSec/helpers/insightAppSecApi.ts

```diff
@@ -285,51 +285,56 @@ export default class InsightAppSecApi
         return new Promise(function (resolve, reject)
         {
             try
-            {
-                var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
-                let xhr = new XMLHttpRequest();
+            {   const axios = require('axios').default;
+
+                let axiosInst = axios.create({
+                    baseURL: endpoint,
+                    headers: {'X-Api-Key': this.apiKey,
+                              'Content-Type': 'application/json',
+                              'Accept': 'application/json',
+                              'User-Agent': "r7:insightappsec-azure-devops-extension/1.0.7"},
+                    responseType: 'text',
+                    transformResponse: [data => data]
+                });
 
-                xhr.open(requestType, endpoint);
-                xhr.setRequestHeader("X-Api-Key", this.apiKey);
-                xhr.setRequestHeader("Content-Type", "application/json");
-                xhr.setRequestHeader("Accept", "application/json");
-                xhr.setRequestHeader("User-Agent", "r7:insightappsec-azure-devops-extension/1.0.6");
 
                 if (payload != null && payload != "")
                 {
-                    xhr.send(JSON.stringify(payload));
+                    payload = JSON.stringify(payload);
                 }
                 else
                 {
-                    xhr.send();
+                    payload = null;
                 }
 
-                xhr.onerror = function()
-                {
+                axiosInst({
+                    method: requestType,
+                    data: payload
+                })
+                .then((response) => {
+                        // Ensure valid status code response
+                        if (response.status < 200 || response.status > 299) {
+                            console.error("Failed to return valid response from InsightAppSec API; Status Code: " + response.status +
+                                ". Please Contact Rapid7 Support if this continues to occur.");
+                            console.error("IAS Error response: " + response.data);
+                            resolve(null);
+                            return;
+                        }
+    
+                        var locationHeader = response.headers["location"];
+    
+                        if (locationHeader != null)
+                        {
+                            var scanId = locationHeader.split("/").pop();
+                            resolve(scanId);
```

Comment:
After we resolve I believe execution continues, so we need to add a return otherwise the rest of the  method's code will execute (in this case resolve(response.data)) will won't have any effect since you cant change a settled promise.